client/my-sites/checkout/src/hooks/use-checkout-v2.ts

```diff
@@ -0,0 +1,23 @@
+import { useExperiment } from 'calypso/lib/explat';
+import { useSelector } from 'calypso/state';
+import { getSectionName } from 'calypso/state/ui/selectors';
+
+export function useCheckoutV2(): 'loading' | 'treatment' | 'control' {
+	const isCheckoutSection = useSelector( getSectionName ) === 'checkout';
+
+	const [ isLoadingExperimentAssignment, experimentAssignment ] = useExperiment(
+		'calypso_launch_checkout_v2',
+		{ isEligible: isCheckoutSection }
+	);
+
```

Comment:
If we also call `hasCheckoutVersion('2')` in this function, then we can remove it everywhere else.

Something like:

```
if ( hasCheckoutVersion('2') ) {
  return 'treatment';
}
```